lib/src/common/github.dart

```diff
@@ -219,22 +219,29 @@ class GitHub {
   /// The future will pass the object returned from this function to the then method.
   /// The default [convert] function returns the input object.
   /// [body] is the data to send to the server.
-  Future<T> postJSON<S, T>(String path,
-          {int statusCode,
-          void fail(http.Response response),
-          Map<String, String> headers,
-          Map<String, String> params,
-          JSONConverter<S, T> convert,
-          String body,
-          String preview}) =>
-      _requestJson('POST', path,
-          statusCode: statusCode,
-          fail: fail,
-          headers: headers,
-          params: params,
-          convert: convert,
-          body: body,
-          preview: preview);
+  /// [S] represents the input type.
+  /// [T] represents the type return from this function after conversion
+  Future<T> postJSON<S, T>(
+    String path, {
+    int statusCode,
+    void fail(http.Response response),
+    Map<String, String> headers,
+    Map<String, String> params,
+    JSONConverter<S, T> convert,
+    dynamic body,
```

Comment:
Hmm, I'm actually concerned about making this dynamic. It's not a breaking change, but it also means that people don't get any help as to what is an allowed type to pass in .. especially for binary uploads. I wonder if it would work to keep the original with a String type and an essentially a copy of `postJSON` called `postJSONBinary` or something that takes the right type for uploading a binary body. The internal `_requestJson` can probably stay dynamic.
What do you think?